sdk/identity/azure-identity/src/environment_credential.cpp

```diff
@@ -15,26 +20,48 @@ using Azure::Core::Credentials::AccessToken;
 using Azure::Core::Credentials::AuthenticationException;
 using Azure::Core::Credentials::TokenCredentialOptions;
 using Azure::Core::Credentials::TokenRequestContext;
+using Azure::Core::Diagnostics::Logger;
+using Azure::Core::Diagnostics::_internal::Log;
+
+namespace {
+auto const AzureTenantIdEnvVarName = "AZURE_TENANT_ID";
```

Comment:
What does adding `const` bring to the table? I'm not aware of any guidelines which recommend const for local variables (parameters is a different story).

And in this case, why not make it `constexpr` instead of `const`?

Reply:
const is good, because it prevents the code from accidentally changing it, and helps reading, but I agree, constexpr is better. I'll change it.